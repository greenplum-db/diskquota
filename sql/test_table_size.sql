-- Test tablesize table

create table a(i text);

insert into a select * from generate_series(1,10000);

select pg_sleep(2);
create table buffer(oid oid, relname name, size bigint);

with size as ( select oid,relname,pg_total_relation_size(oid) from pg_class) insert into buffer select size.oid, size.relname, size.pg_total_relation_size  from size, diskquota.table_size as dt where dt.tableid = size.oid and relname = 'a';

insert into buffer select oid, relname, sum(pg_total_relation_size(oid)) from gp_dist_random('pg_class') where oid > 16384 and (relkind='r' or relkind='m') and relname = 'a' group by oid, relname;

select sum(buffer.size) = diskquota.table_size.size from buffer, diskquota.table_size where buffer.oid = diskquota.table_size.tableid group by diskquota.table_size.size;
