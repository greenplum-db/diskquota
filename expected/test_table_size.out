-- Test tablesize table
create table a(i text);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
insert into a select * from generate_series(1,10000);
select pg_sleep(2);
 pg_sleep 
----------
 
(1 row)

create table buffer(oid oid, relname name, size bigint);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'oid' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
with size as ( select oid,relname,pg_total_relation_size(oid) from pg_class) insert into buffer select size.oid, size.relname, size.pg_total_relation_size  from size, diskquota.table_size as dt where dt.tableid = size.oid and relname = 'a';
insert into buffer select oid, relname, sum(pg_total_relation_size(oid)) from gp_dist_random('pg_class') where oid > 16384 and (relkind='r' or relkind='m') and relname = 'a' group by oid, relname;
select sum(buffer.size) = diskquota.table_size.size from buffer, diskquota.table_size where buffer.oid = diskquota.table_size.tableid group by diskquota.table_size.size;
 ?column? 
----------
 t
(1 row)

